### ISSUE TITLE
`rvm install <version>` fails to regenerate gem wrappers when `ls` has been aliased in certain ways

### ISSUE THREAD

**Issue by rdodson41** [rvm/rvm#4282]
## Description

`rvm install <version>` fails to regenerate gem wrappers when `ls` has been aliased in certain ways.

## Steps to reproduce

 1. Alias `ls` as follows:

```
$ alias ls='"/usr/local/opt/coreutils/libexec/gnubin/ls" --format=long --human-readable --time-style=long-iso --classify --color=auto'
```

 2. Run `rvm install ruby-2.4.3`.

## Expected behavior

* [`rvm install ruby-2.4.3`](https://gist.github.com/rdodson41/53e0a382e69045b55ee17048c9770f28)

Note the relevant successful steps:

```
ruby-2.4.3 - #importing gemset /Users/rich/.rvm/gemsets/global.gems.............................................
ruby-2.4.3 - #generating global wrappers.......
```

## Actual behavior

* [`rvm install ruby-2.4.3`](https://gist.github.com/rdodson41/d62f413b0f8220ce5e79b28864a5a8c9)

Note the relevant errors when `rvm` attempts to regenerate gem wrappers:

```
ruby-2.4.3 - #importing gemset /Users/rich/.rvm/gemsets/global.gems.............there was an error installing gem gem-wrappers
.................................there was an error installing gem rvm
.
ruby-2.4.3 - #generating global wrappers.............
Error running 'run_gem_wrappers regenerate',
please read /Users/rich/.rvm/log/1516069374_ruby-2.4.3/gemset.wrappers.global.log
```

* [`gemset.wrappers.global.log`](https://gist.github.com/rdodson41/a81a9ce6f012488ea3a2d2a03e3c7a28)

Note the relevant errors:

```
++ printf %b 'command(7): command gem install -rw-r--r-- 1 rich staff 13K 2018-01-15 21:20 /Users/rich/.rvm/gem-cache/gem-wrappers-1.4.0.gem --local --no-ri --no-rdoc\n'
++ rvm_log 'installing gem -rw-r--r-- 1 rich staff 13K 2018-01-15 21:20 /Users/rich/.rvm/gem-cache/gem-wrappers-1.4.0.gem --local --no-ri --no-rdoc'
++ [[ '' == 1 ]]
++ printf %b 'installing gem -rw-r--r-- 1 rich staff 13K 2018-01-15 21:20 /Users/rich/.rvm/gem-cache/gem-wrappers-1.4.0.gem --local --no-ri --no-rdoc\n'
++ command gem install '-rw-r--r-- 1 rich staff 13K 2018-01-15 21:20 /Users/rich/.rvm/gem-cache/gem-wrappers-1.4.0.gem' --local --no-ri --no-rdoc
++ gem install '-rw-r--r-- 1 rich staff 13K 2018-01-15 21:20 /Users/rich/.rvm/gem-cache/gem-wrappers-1.4.0.gem' --local --no-ri --no-rdoc
ERROR:  While executing gem ... (OptionParser::InvalidOption)
    invalid option: - 1 rich staff 13K 2018-01-15 21:20 /Users/rich/.rvm/gem-cache/gem-wrappers-1.4.0.gem
```

* I suspect that the following line is responsible for this issue:

[`__available_gem="$( ls -v1 "${rvm_path}/gem-cache"/${gem_name}-${version_check}.gem 2>/dev/null | tail -n 1 )"`](https://github.com/rvm/rvm/blob/master/scripts/functions/gemset#L169)

## Environment info

* [`rvm info`](https://gist.github.com/rdodson41/921a2f3b50a36d25bd8766ade14e4b9a)

**Comment by skunkworker** [comment]
@pkuczynski I'm seeing this behavior currently on zsh when using the alias alias ls='lsd'. Should I open another issue or use this one?

**Comment by pkuczynski** [comment]
So the fix in #4283 does not work for you? Ideally, you might try to come up with a PR :)

**Comment by skunkworker** [comment]
@pkuczynski I've got a tentative PR that works, I would love some feedback on the direction though as my strategy may not cover all that it needs to. #4743
